src/components/RequestPanel/RequestStatus.js

```diff
@@ -0,0 +1,282 @@
+            `}
+          />
+        ) : (
+          <IconCheck
+            size="medium"
+            css={`
+              color: ${iconColor};
+              transition: color 150ms ease-in-out;
+            `}
+          />
+        )}
+      </div>
+      <p
+        css={`
+          margin-top: ${3 * GU}px;
+        `}
+      >
+        {label}
+      </p>
+    </div>
+  )
+}
+
+function RequestStatusInfo({
+  allSuccess,
+  multipleRequests,
+  maxAttemptsReached,
+  onClosePanel,
+  onNextAttempt,
+  requests,
+  requestStatus,
+}) {
+  const { activated } = useWallet()
+  const provider = getProviderFromUseWalletId(activated)
+
+  if (allSuccess) {
+    return (
+      <Info>Request{multipleRequests ? 's' : ''} processed successfully!</Info>
+    )
+  }
+
+  const requiresMultipleSignatures =
+    multipleRequests && requests.filter(request => request.isTx).length > 1
+  const requestFailedStatus = requestStatus.find(status =>
+    [REQUEST_STATUS_PROCESSING_FAILED, REQUEST_STATUS_FAILED].includes(status)
+  )
+
+  if (requestFailedStatus) {
+    if (multipleRequests) {
+      return (
+        <RequestReattempt
+          maxAttemptsReached={maxAttemptsReached}
+          onClosePanel={onClosePanel}
+          onNextAttempt={onNextAttempt}
+        />
+      )
+    }
+
+    return (
+      <>
+        <Info>
+          {requestFailedStatus === REQUEST_STATUS_PROCESSING_FAILED
+            ? "Your request wasn't processed correctly"
+            : 'Your request failed'}
+        </Info>
+        <Button
+          css={`
+            margin-top: ${2 * GU}px;
+          `}
+          label="Close"
+          mode="strong"
+          onClick={onClosePanel}
+          wide
+        />
+      </>
+    )
+  }
+
+  if (multipleRequests) {
+    return (
+      <Info>
+        {requiresMultipleSignatures ? (
+          <span>
+            This action requires multiple transactions. Please sign them one
+            after another and <b>do not close this window</b> until the process
+            is finished. Open your Ethereum provider {provider.name} to sign the
+            transactions.{' '}
+          </span>
+        ) : (
+          <span>
+            This action requires multiple steps. Please,{' '}
+            <b>do not close this window</b> until the process is finished.
+          </span>
+        )}
+      </Info>
+    )
+  }
+
+  // If we get here means we have only one request
+  return (
+    <Info>
+      {requests[0].isTx
+        ? `Open ${provider.name} to sign your transaction`
+        : requests[0].description}
+    </Info>
+  )
+}
+
+const RequestReattempt = ({
+  maxAttemptsReached,
+  onClosePanel,
+  onNextAttempt,
+}) => {
+  return (
+    <>
+      <Info>
+        {maxAttemptsReached
+          ? `Seems that the request can't be processed`
+          : `An error has occurred during the request process. Don't worry, you can
+             try to process the request again.`}
```

Comment:
```suggestion
          : `An error occurred at this step. Don't worry, you can try to process
             this step again.`}
```
We could probably improve this... but let's not worry about it for now.